src/webgl/p5.RendererGL.js

```diff
@@ -872,12 +877,51 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
     this.curStrokeJoin = join;
   }
 
-  filter(filterType) {
-    // filter can be achieved using custom shaders.
-    // https://github.com/aferriss/p5jsShaderExamples
-    // https://itp-xstory.github.io/p5js-shaders/#/
-    p5._friendlyError('filter() does not work in WEBGL mode');
+  filter(args) {
+    // Couldn't create graphics in RendererGL constructor
+    // (led to infinite loop)
+    // so it's just created here once on the initial filter call.
+    if (!this.filterGraphicsLayer) {
+      this.filterGraphicsLayer =
+        new p5.Graphics(this.width, this.height, constants.WEBGL, this._pInst);
```

Comment:
Could it not be working with pgraphics because this._pInst is referring to the main sketch?

Reply:
Yeah I think that's it. Would the alternative be making a new `p5` instance there?

Here's some logs:
![image](https://github.com/processing/p5.js/assets/28441593/87168f61-1eeb-4d3e-a520-9fa775d60620)

Reply:
Also trying to reference the other time in the source where a p5.Graphics is created, in `src/core/rendering.js`:
```
p5.prototype.createGraphics = function(w, h, renderer, canvas) {
  ...
  return new p5.Graphics(w, h, renderer, this, canvas);
}
```